CMakeLists.txt

```diff
@@ -0,0 +1,148 @@
+cmake_minimum_required(VERSION 3.13)
+# add_link_options requires 3.13 https://cmake.org/cmake/help/v3.13/command/add_link_options.html
+
+project(qgrep)
+
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+
+if (WIN32)
+    add_compile_options(/DUSE_SSE2 /DNOMINMAX /wd"4996" /wd"4267" /wd"4244")
+
+    if (CMAKE_SYSTEM_PROCESSOR MATCHES "(x86)|(X86)")
+        message(STATUS "use SSE2 of ${CMAKE_SYSTEM_PROCESSOR}")
+        add_compile_options(/arch:SSE2)
+    else()
+        message(STATUS "MSVC on ${CMAKE_SYSTEM_PROCESSOR} has /arch:SSE2 by default")
+    endif ()
+else()
+    add_compile_options(-c -g -Wall -Werror -O3 -mtune=native -MMD -MP)
```

Comment:
I think you can omit `-c`, `-g` and `-MMD` / `-MP` as these I'd expect to be set up by CMake internally. I'd also omit `-O3` as CMake will do this automatically when using Release configuration.

Reply:
You are right. I am removing them and use `RelWithDebInfo` as `CMAKE_BUILD_TYPE` default value.

However, in this case, CMake will pass `-DNDEBUG` to compiler, which disables `assert()`. This leads to several errors (`-Werror=unused-variable`) in `src/compression.cpp` and `src/encoding.cpp`. The reason is:

```
int result = convertToUTF8Impl or LZ4_decompress_safe (...)
assert(result ...)
```
`assert` is removed by `-DNDEBUG` and `result` becomes unused variable. I will fix them too.